game-app/game-core/src/main/java/games/strategy/engine/framework/startup/launcher/ServerLauncher.java

```diff
@@ -271,6 +271,12 @@ public void connectionLost(final INode node) {
     if (serverGame.getPlayerManager().isPlaying(node)) {
       if (serverGame.isGameSequenceRunning()) {
         saveAndEndGame(node);
+        // Release any countries played by this player. This ensures the slots are freed up in the
+        // game lobby as we keep the other player mappings for resuming from the saved game.
+        for (final String countryName : serverGame.getPlayerManager().getPlayedBy(node)) {
+          serverModel.releasePlayer(countryName);
+        }
```

Comment:
What are your thoughts if we changed the initial seat allocation code to not occupy seats by the host? Is there a complexity there, are you striving for avoiding changing existing behavior?

I like the idea of simply not allocating those seats if it means we can avoid this for loop entirely. This code relies on an ordering of execution that could be brittle. Thereby if we moved the seat allocation code to execute after this, we would have an error (which also speaks to the overall complexity of this whole scenario, I acknowledge working through that would have been difficult! There is a lot of state a person needs to grok to follow the server launcher logic, it's not easy)

Reply:
Sorry, I'm not following this comment.

This logic is not about "seats occupied by the host" - this is looping through seats occupied by the player who just dropped. Basically, they dropped, so clear their seats.

Previously, this wasn't needed because a restart would just re-initialize the seats. I guess maybe you're asking if we could just change that code instead to not do that? That seems like a reasonable behavior change to me, but seems orthogonal to this change.

And more specifically, even if that were the case, then we'd still need to have this loop to clear seats when someone drops.

I guess one thing such a change could do is make it so we don't need to have a special `playersToNodesMappingPersisted` codepath - because we would just always persist mappings and only clear them when a map changes.

I think that could be an option, but would be a bigger change.

Reply:
Oh, I was confused about what was happening. I thought `node` was the host. In `ServerLauncher` and `ServerModel`, there is a reference to the 'host node' that is typically referred to as just 'the node'. (Which is all kinda crazy naming terminology, but c'est la vie).

It seems we can bring some additional clarity here. This would be a great comment on the for loop: "this is looping through seats occupied by the player who just dropped. Basically, they dropped, so clear their seats."

Second, renaming `node` to something like `nodeThatDropped` or something like. (More ideally we'd call it `playerThatDropped` but unfortunately the code naming is f%#@d up and 'players' are called 'nodes'. So presumably that terminology should stay consistent in the commentary too).

Reply:
Done, renamed to `droppedNode` (to keep more concise that your suggestion which would have caused extra wrapping) and updated the comment to make it clearer.